add examples for distinct on (#27087)
dimagi/commcare-hq: corehq/apps/userreports/README.rst

     }
   ]

Distinct On

> -----------

Reviewer: ```suggestion
-----------
Can be used to limit the rows in a report to a single unique column or set of columns, even if the underlying
query would return more than one result.

```

Based on columns mentioned in the distinct on clause,

> The top most row is picked in case of duplicates.

Reviewer: ```suggestion
the top most row is picked in case of duplicates.
```

This is different from aggregation in sense that this is done after fetching
the rows, whereas aggregation is done before selecting the rows.

> This is used in combination with a sort expression to have predictable results.

Reviewer: ```suggestion
This should be used in combination with a sort expression to have predictable results, otherwise an arbitrary
row would be returned first.
```

Author: this can't be used without a sort expression and the reason is to have predictability.
`is` to `should be` sounds the same?
there is no otherwise since it cant be used without it, yes?

Reviewer: gotcha, I didn't realize that. thanks.

> Please note that the columns used in distinct on clause should also be present

Reviewer: ```suggestion
Therefore, the columns used in distinct on clause should also be present
```

in the sort expression as the first set of columns in the same order.

Pick distinct by a single column
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Sort expression should have username and then other columns if needed

.. code:: json

   [
     {
       "field": "column1",
       "order": "DESC"
     },
     {
       "field": "column2",
       "order": "ASC"
     }
   ]

distinct on would be

.. code:: json

   ["column1"]

> 

Reviewer: ```suggestion

The underlying report would then show a single row for each distinct value in `"column1"`.
```
